SpatialGDK/Source/SpatialGDK/Public/Utils/RepLayoutUtils.h

```diff
@@ -188,4 +188,23 @@ inline TArray<UFunction*> GetClassRPCFunctions(const UClass* Class)
 	return RelevantClassFunctions;
 }
 
+inline bool IsFastArraySerializeProperty(UArrayProperty* Property, UProperty* ParentProperty)
+{
+	if (UStructProperty* ParentStruct = Cast<UStructProperty>(ParentProperty))
```

Comment:
Could the outmost if be removed?

Is it ever the case that the parent struct could be a fast array serializer and the inner one isn't? If not, could this be condensed to just checking `Property->Inner`?

Reply:
Not tailored just for GAS, but FastArraySerialization. To use that functionality it requires a parent struct which extends from FFastArraySerializer, with an internal member array who's data type extends from FFastArraySerializerItem. This is the relationship I'm looking for. Will add a comment.

Reply:
Right, but this won't work if the parent struct of the array is a property on another struct, and not on the actor/object itself, since `ParentProperty` here will actually be the top level property (the one marked with `Replicated`)

Reply:
Reworked this logic.